### PR TITLE
Change guidance about dictionaries to cover all optional arguments

### LINKED ISSUE #366: New principle: Prefer dictionary arguments for all optional parameters (and non-obvious parameters?)
Right now we have weaker guidance: [6.4. Prefer dictionary arguments over primitive arguments](https://w3ctag.github.io/design-principles/#prefer-dict-to-bool). However, I think dictionary arguments are preferable even for non-primitive arguments, if they are optional or non-obvious by looking at their value. E.g. consider `fetch()`, imagine if `headers` was positional simply because it's an object.

Dictionary arguments are preferable because:
- They are named, so what each parameter does is obvious
- They can be provided in any order
- They make the API more extensible in the future

WDYT?

### PR DESCRIPTION
**Pull request by LeaVerou**
Closes #366

***
<a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/pull/416.html" title="Last updated on Feb 8, 2023, 1:09 AM UTC (7514f7e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/416/a35df00...7514f7e.html" title="Last updated on Feb 8, 2023, 1:09 AM UTC (7514f7e)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -714,7 +714,7 @@ following categories:
 Specify how your feature works
 with non-[=Document/fully active=] BFCached documents,
 following the guidelines in the
-[Supporting BFCached Documents](https://w3ctag.github.io/bfcache-guide/) guide.  
+[Supporting BFCached Documents](https://w3ctag.github.io/bfcache-guide/) guide.
```
Comment (cynthia) [inline index.bs:717]:
```suggestion
[Supporting BFCached Documents](https://w3ctag.github.io/bfcache-guide/) guide.  
```

File: index.bs [hunk 2]
```diff
@@ -892,7 +892,7 @@ and because the <{img}> element is an empty element that does not permit child e
 A space-separated syntax would not have been possible, as each list item includes multiple values.
 
 <!--
-TODO: link to the "microsyntaxes are usually bad" principle once it's been written. 
+TODO: link to the "microsyntaxes are usually bad" principle once it's been written.
```
Comment (cynthia) [inline index.bs:895]:
```suggestion
TODO: link to the "microsyntaxes are usually bad" principle once it's been written. 
```

File: index.bs [hunk 3]
```diff
@@ -2584,7 +2590,7 @@ APIs which would often block the main thread if used as intended
 should not be exposed on the {{Window}} interface.
 By restricting such APIs to the {{DedicatedWorkerGlobalScope}} interface,
 the “easy” path for web developers
-is the path with the best experience for users. 
+is the path with the best experience for users.
```
Comment (cynthia) [inline index.bs:2593]:
```suggestion
is the path with the best experience for users. 
```

**Review by cynthia — APPROVED** [review]

File: index.bs [hunk 4]
```diff
@@ -1394,22 +1394,21 @@ This means:
 - If static, then return a new object each time.
     In which case, this should be be a method.
 
-<h3 id="prefer-dict-to-bool">Prefer dictionary arguments over primitive arguments</h3>
+<h3 id="prefer-dict-to-bool">Accept optional and/or primitive arguments through dictionaries</h3>
```
Comment (LeaVerou) [inline index.bs:1397]:
```suggestion
<a  id="prefer-dict-to-bool"></a>
<h3 id="prefer-dictionaries">Accept optional and/or primitive arguments through dictionaries</h3>
```

File: index.bs [hunk 5]
```diff
@@ -1418,7 +1417,14 @@ is much more readable than
 new Event("example", true, false)
 </pre>
 
-Also,
+</div>
+
+API designers should also consider
```
Comment (cynthia) [inline index.bs:1422]: You should use active tone (following the style guide) here - e.g. You should consider accepting parameters through a dictionary.

Comment (LeaVerou) [inline index.bs:1422]:
```suggestion
You should also consider
```

File: index.bs [hunk 6]
```diff
@@ -1418,7 +1420,14 @@ is much more readable than
 new Event("example", true, false)
 </pre>
 
-Also,
+</div>
+
+You should consider
```
Comment (LeaVerou) [inline index.bs:1425]:
```suggestion
You should also consider
```

**Pull request merged**